src/main/java/com/shopify/ShopifySdk.java

```diff
@@ -645,6 +597,75 @@ public ShopifyCustomer updateCustomer(final ShopifyCustomerUpdateRequest shopify
 		return shopifyCustomerRootResponse.getCustomer();
 	}
 
+	public ShopifyCustomer getCustomerById(final String id) {
+		final Response response = get(getWebTarget().path(CUSTOMERS).path(id));
+		final ShopifyCustomerRoot shopifyCustomerRootResponse = response.readEntity(ShopifyCustomerRoot.class);
+		return shopifyCustomerRootResponse.getCustomer();
+	}
+
+	public List<ShopifyCustomer> getCustomers(Response response) {
+		ShopifyCustomersRoot shopifyCustomersRootResponse = response.readEntity(ShopifyCustomersRoot.class);
+		return shopifyCustomersRootResponse.getCustomers();
+	}
+
+	public List<ShopifyCustomer> getCustomers(final int page) {
+		return getCustomers(page, DEFAULT_REQUEST_LIMIT);
+	}
+
+	public List<ShopifyCustomer> getCustomers(final int page, final int pageSize) {
+		final Response response = get(getWebTarget().path(CUSTOMERS)
+				.queryParam(LIMIT_QUERY_PARAMETER, pageSize)
+				.queryParam(PAGE_QUERY_PARAMETER, page));
+		return getCustomers(response);
+	}
+
+	public List<ShopifyCustomer> getCustomers(List<String> ids) {
+		final Response response = get(getWebTarget().path(CUSTOMERS)
+				.queryParam(IDS_QUERY_PARAMETER, StringUtils.join(ids, ",")));
+		return getCustomers(response);
+	}
+
+	public List<ShopifyCustomer> getCustomers(String sinceId) {
+		final Response response = get(getWebTarget().path(CUSTOMERS)
+				.queryParam(SINCE_ID_QUERY_PARAMETER, sinceId));
+		return getCustomers(response);
+	}
+
+	public List<ShopifyCustomer> getCustomers(final DateTime mininumCreationDate, final int page, final int pageSize) {
```

Comment:
How do you feel about consolidating some of these getCustomers function that take in some of the similar parameters with a single model that they'd use a step builder to create? We can kind of guide end users into creating the correct params. You can view some of our other models that use the step builder.

I'd like your opinion. I'm kind of torn because that could result in a larger method, but be a little bit easier to maintain and support.

Reply:
I really like the idea of using step builder -- it'd be good to move all the orders-related calls to that pattern as well (maybe not in this PR, but sometime in the future), because I can definitely see the use case there too. My only concern is adding a lot of code to an already large file. I worry that implementing another step builder in `ShopifySdk` would make the file a bit harder to read. Do you think there's a way to move it to its own file?

Reply:
Oh, I actually totally see what you mean now having looked through the models again. I can definitely do that.